clang/lib/Sema/SemaInit.cpp

```diff
@@ -4561,6 +4569,15 @@ static void TryConstructorInitialization(Sema &S,
 
   CXXConstructorDecl *CtorDecl = cast<CXXConstructorDecl>(Best->Function);
   if (Result != OR_Deleted) {
+    if (!IsListInit && Kind.getKind() == InitializationKind::IK_Default &&
+        DestRecordDecl != nullptr && DestRecordDecl->isAggregate() &&
+        DestRecordDecl->hasUninitializedExplicitInitFields()) {
+      S.Diag(Kind.getLocation(), diag::warn_field_requires_explicit_init)
+          << ("in " +
```

Comment:
"in " should be part of the diagnostic, we shouldn't be calculating strings like this here (as the idea is that some day we should be able to translate the DiagnosticSemaKinds.td file for a translated compiler).

Reply:
Check out the `select{whatever | something else}0` form of diagnostics.